src/import/ImportRaw.cpp

```diff
@@ -288,31 +274,32 @@ void ImportRaw(const AudacityProject &project, wxWindow *parent, const wxString
 
 enum {
    ChoiceID = 9000,
-   PlayID
+   PlayID,
+   DetectID,
 };
 
 BEGIN_EVENT_TABLE(ImportRawDialog, wxDialogWrapper)
    EVT_BUTTON(wxID_OK, ImportRawDialog::OnOK)
    EVT_BUTTON(wxID_CANCEL, ImportRawDialog::OnCancel)
    EVT_BUTTON(PlayID, ImportRawDialog::OnPlay)
+   EVT_BUTTON(DetectID, ImportRawDialog::OnDetect)
    EVT_CHOICE(ChoiceID, ImportRawDialog::OnChoice)
 END_EVENT_TABLE()
 
-ImportRawDialog::ImportRawDialog(wxWindow * parent,
-                                 int encoding, unsigned channels,
-                                 int offset, double rate)
+ImportRawDialog::ImportRawDialog(wxWindow * parent, const wxString & fileName)
 :  wxDialogWrapper(parent, wxID_ANY, XO("Import Raw Data"),
             wxDefaultPosition, wxDefaultSize,
             wxDEFAULT_DIALOG_STYLE | wxRESIZE_BORDER),
-    mEncoding(encoding),
-    mChannels(channels),
-    mOffset(offset),
-    mRate(rate)
+   mFileName(fileName)
 {
-   wxASSERT(channels >= 1);
+   wxASSERT(mChannels >= 1);
 
    SetName();
 
+   // Append filename at window title
+   wxFileName wfn{ fileName };
+   wxDialog::SetTitle(GetTitle() + ": " + wfn.GetFullName());
```

Comment:
Make the title
```
XO("%s: %s").Format(GetTitle(), wfn.GetFullName()).Translation()
```

This is the substitute-don't-concatenate rule for user-visible strings.

And yes, even that string might be localized differently!  For instance French prefers a space left of the colon as well.

Reply:
Done